File: test_archives/py_tests/batch5/test_function2.py
from functions.function2 import function2
import pytest

@pytest.fixture
def function2_obj():
    return function2("John", 12, 4, "English")

def test_function2_name(function2_obj):
    assert function2_obj.name == "John"

def test_function2_age(function2_obj):
    assert function2_obj.age == 12

def test_function2_grade(function2_obj):
    assert function2_obj.grade == 4

def test_function2_language(function2_obj):
    assert function2_obj.language == "English"

def test_function2_options(function2_obj):
    function2_obj.set_options(True, False, True, False)
    assert function2_obj.option1 == True
    assert function2_obj.option2 == False
    assert function2_obj.option3 == True
    assert function2_obj.option4 == False

File: test_archives/cpp_tests/batch5/test_function2.cpp
#include "../catch.hpp"
#include "../functions/function2.cpp"

TEST_CASE("test function2 name", "[function2]") {
    function2 function2_obj ("John", 12, 4, "English");
    REQUIRE(function2_obj.name == "John");
}

TEST_CASE("test function2 age", "[function2]") {
    function2 function2_obj ("John", 12, 4, "English");
    REQUIRE(function2_obj.age == 12);
}

TEST_CASE("test function2 grade", "[function2]") {
    function2 function2_obj ("John", 12, 4, "English");
    REQUIRE(function2_obj.grade == 4);
}

TEST_CASE("test function2 language", "[function2]") {
    function2 function2_obj ("John", 12, 4, "English");
    REQUIRE(function2_obj.language == "English");
}

TEST_CASE("test function2 options", "[function2]") {